lldb/include/lldb/Core/ValueObject.h

```diff
@@ -441,6 +441,19 @@ class ValueObject {
 
   virtual int64_t GetValueAsSigned(int64_t fail_value, bool *success = nullptr);
 
+  llvm::APSInt GetValueAsAPSInt();
+
+  llvm::APFloat GetValueAsFloat();
+
+  bool GetValueAsBool();
+
+  /// Update the value of the current object to be the integer in the 'value'
+  /// parameter.
+  void UpdateIntegerValue(const llvm::APInt &value);
+
+  /// Assign the integer value 'new_val_sp' to the current object.
+  void UpdateIntegerValue(lldb::ValueObjectSP new_val_sp);
```

Comment:
Yes, this assigns the value from new_val_sp to the current (integer) ValueObject.  The main part of this function does call the preceding one. Because new_val_sp can actually contain an integer, a float or a pointer, there's some slightly different manipulations of the value that needs to happen before calling the preceding function (take a look at the code in ValueObject.cpp), which is why this function was written.